dawn/src/dawn/Serialization/ASTSerializer.cpp

```diff
@@ -136,13 +144,25 @@ void setAccesses(dawn::proto::statements::Accesses* protoAccesses,
 }
 
 iir::Extents makeExtents(const dawn::proto::statements::Extents* protoExtents) {
-  int dim1minus = protoExtents->extents()[0].minus();
-  int dim1plus = protoExtents->extents()[0].plus();
-  int dim2minus = protoExtents->extents()[1].minus();
-  int dim2plus = protoExtents->extents()[1].plus();
-  int dim3minus = protoExtents->extents()[2].minus();
-  int dim3plus = protoExtents->extents()[2].plus();
-  return {ast::cartesian, dim1minus, dim1plus, dim2minus, dim2plus, dim3minus, dim3plus};
+  iir::Extent vExtent{protoExtents->vertical_extent().minus(),
+                      protoExtents->vertical_extent().plus()};
+
+  if(protoExtents->has_cartesian_extent()) {
+    return {iir::HorizontalExtent{ast::cartesian,
+                                  protoExtents->cartesian_extent().i_extent().minus(),
+                                  protoExtents->cartesian_extent().i_extent().plus(),
+                                  protoExtents->cartesian_extent().j_extent().minus(),
+                                  protoExtents->cartesian_extent().j_extent().plus()},
+            vExtent};
+  } else if(protoExtents->has_unstructured_extent()) {
+    return {
+        iir::HorizontalExtent{ast::unstructured, protoExtents->unstructured_extent().has_extent()},
+        vExtent};
+  } else if(protoExtents->has_zero_extent()) {
```

Comment:
This is confusing. It should be `has_zero_horizontal_extent` or similar. Not sure if the other need to be explicit as well (`has_unstructured_horizontal_extent`), probably not.

Reply:
I tried to improve a bit. Please have a look.